unified-codes/apps/data-service/src/app/types/DgraphDataSource.ts

```diff
@@ -64,14 +65,24 @@ export class DgraphDataSource extends RESTDataSource {
         total: count(uid)
       }
       
-      query(func: uid(all), ${orderString}, offset: ${offset}, first: ${first}) @recurse(loop: false)  {
+      query(func: uid(all), ${orderString}, offset: ${offset}, first: ${first})  {
         code
         description
         type
-        uid
-        value
-        children: has_child
-        properties: has_property
+        properties: has_property {
+          type
+          value
+        }
+        parent: ~has_child {
+          parent:  ~has_child {
+            parent: ~has_child {          
+              properties: has_property {
+                type
+                value
+              }
+            }
+          }
+        }
```

Comment:
I think there's a more elegant way to do this in dgraph by using `@cascade`  (allows you to "bubble up" filters at lower levels to higher levels, i.e. "get properties for node which has child with code `x`"). Think best is if I do a demo in an issue rather than request changes to the PR 😨 .